### PR TITLE
fix: Use non-root user to install Python packages and use a venv in CUDA Docker image

### PR DESCRIPTION
**Pull request by emre-aydin**
# The problem

Root user is used for installing Python module and the system Python executable is used, which creates problems for custom image authors.

# This PR's solution

This is a follow-up PR to another and makes the same change for the CUDA images.

# Jira ticket

https://zapatacomputing.atlassian.net/browse/OP-166

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: docker/cuda.Dockerfile [hunk 1]
```diff
@@ -35,25 +36,30 @@ ENV CUQUANTUM_DIR=/opt/nvidia/cuquantum
 # get required tools to build qsim
 RUN <<EOF
 python -m pip install pybind11
```
Comment (alexjuda) [inline docker/cuda.Dockerfile:38]: It looks like this `pip install` happens outside of our venv. Should this line be moved down?

Comment (emre-aydin) [reply docker/cuda.Dockerfile:38]: Yes, AFAIU it's required only for that install so I chose to not pollute our venv with it (I might be mistaken though).

Comment (emre-aydin) [reply docker/cuda.Dockerfile]: @alexjuda after adapting to changes in https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/233, this is now being installed in the same virtualenv.

**Review by roryschadler — APPROVED** [review]

**Review by alexjuda — APPROVED** [review]
I'll rely on @roryschadler's approval wrt. alignment with the other docker images discussed this week! Apart from it, LGTM

**Pull request merged**